### ISSUE TITLE
Could not build the wheels for 0.2.86 on Mac Silicon

### ISSUE THREAD

**Issue by mobeetle** [abetlen/llama-cpp-python#1663]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ x ] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [ x ] I carefully followed the [README.md](https://github.com/abetlen/llama-cpp-python/blob/main/README.md).
- [ x ] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [ x ] I reviewed the [Discussions](https://github.com/abetlen/llama-cpp-python/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

Install llama-cpp-python 0.2.86

# Current Behavior
# Environment and Context

(llama-cpp-python) ~/Documents/CODING/models
CMAKE_ARGS="-DGGML_METAL=on" pip install llama-cpp-python --upgrade --force-reinstall --no-cache-dir
Collecting llama-cpp-python
  Downloading llama_cpp_python-0.2.86.tar.gz (49.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 49.3/49.3 MB 22.1 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Collecting typing-extensions>=4.5.0 (from llama-cpp-python)
  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting numpy>=1.20.0 (from llama-cpp-python)
  Downloading numpy-2.0.1-cp312-cp312-macosx_14_0_arm64.whl.metadata (60 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.9/60.9 kB 36.2 MB/s eta 0:00:00
Collecting diskcache>=5.6.1 (from llama-cpp-python)
  Downloading diskcache-5.6.3-py3-none-any.whl.metadata (20 kB)
Collecting jinja2>=2.11.3 (from llama-cpp-python)
  Downloading jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting MarkupSafe>=2.0 (from jinja2>=2.11.3->llama-cpp-python)
  Downloading MarkupSafe-2.1.5-cp312-cp312-macosx_10_9_universal2.whl.metadata (3.0 kB)
Downloading diskcache-5.6.3-py3-none-any.whl (45 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 45.5/45.5 kB 326.7 MB/s eta 0:00:00
Downloading jinja2-3.1.4-py3-none-any.whl (133 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.3/133.3 kB 48.2 MB/s eta 0:00:00
Downloading numpy-2.0.1-cp312-cp312-macosx_14_0_arm64.whl (5.0 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.0/5.0 MB 31.5 MB/s eta 0:00:00
Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Downloading MarkupSafe-2.1.5-cp312-cp312-macosx_10_9_universal2.whl (18 kB)
Building wheels for collected packages: llama-cpp-python
  Building wheel for llama-cpp-python (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [83 lines of output]
      *** scikit-build-core 0.10.1 using CMake 3.30.2 (wheel)
      *** Configuring CMake...
      loading initial cache file /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/CMakeInit.txt
      -- The C compiler identification is AppleClang 15.0.0.15000309
      -- The CXX compiler identification is AppleClang 15.0.0.15000309
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Found Git: /usr/bin/git (found version "2.39.3 (Apple Git-146)")
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
      -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
      -- Found Threads: TRUE
      -- Accelerate framework found
      -- Metal framework found
      -- The ASM compiler identification is AppleClang
      -- Found assembler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
      -- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
      -- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES)
      -- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND)
      CMake Warning at vendor/llama.cpp/ggml/src/CMakeLists.txt:167 (message):
        OpenMP not found
      
      
      -- Looking for dgemm_
      -- Looking for dgemm_ - found
      -- Found BLAS: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/System/Library/Frameworks/Accelerate.framework
      -- BLAS found, Libraries: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/System/Library/Frameworks/Accelerate.framework
      -- BLAS found, Includes:
      -- Using llamafile
      -- Warning: ccache not found - consider installing it for faster compilation or disable this warning with GGML_CCACHE=OFF
      -- CMAKE_SYSTEM_PROCESSOR: arm64
      -- ARM detected
      -- Performing Test COMPILER_SUPPORTS_FP16_FORMAT_I3E
      -- Performing Test COMPILER_SUPPORTS_FP16_FORMAT_I3E - Failed
      CMake Warning (dev) at CMakeLists.txt:9 (install):
        Target llama has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
      Call Stack (most recent call first):
        CMakeLists.txt:73 (llama_cpp_python_install_target)
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
      CMake Warning (dev) at CMakeLists.txt:17 (install):
        Target llama has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
      Call Stack (most recent call first):
        CMakeLists.txt:73 (llama_cpp_python_install_target)
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
      CMake Warning (dev) at CMakeLists.txt:9 (install):
        Target ggml has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
      Call Stack (most recent call first):
        CMakeLists.txt:74 (llama_cpp_python_install_target)
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
      CMake Warning (dev) at CMakeLists.txt:17 (install):
        Target ggml has PUBLIC_HEADER files but no PUBLIC_HEADER DESTINATION.
      Call Stack (most recent call first):
        CMakeLists.txt:74 (llama_cpp_python_install_target)
      This warning is for project developers.  Use -Wno-dev to suppress it.
      
      -- Configuring done (0.7s)
      -- Generating done (0.0s)
      -- Build files have been written to: /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build
      *** Building project with Ninja...
      Change Dir: '/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build'
      
      Run Build Command(s): /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-build-env-ylemwrpn/normal/lib/python3.12/site-packages/ninja/data/bin/ninja -v
      [1/34] cd /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp && /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-build-env-ylemwrpn/normal/lib/python3.12/site-packages/cmake/data/bin/cmake -DMSVC= -DCMAKE_C_COMPILER_VERSION=15.0.0.15000309 -DCMAKE_C_COMPILER_ID=AppleClang -DCMAKE_VS_PLATFORM_NAME= -DCMAKE_C_COMPILER=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -P /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/common/cmake/build-info-gen-cpp.cmake
      FAILED: /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/common/build-info.cpp
      cd /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp && /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-build-env-ylemwrpn/normal/lib/python3.12/site-packages/cmake/data/bin/cmake -DMSVC= -DCMAKE_C_COMPILER_VERSION=15.0.0.15000309 -DCMAKE_C_COMPILER_ID=AppleClang -DCMAKE_VS_PLATFORM_NAME= -DCMAKE_C_COMPILER=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -P /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/common/cmake/build-info-gen-cpp.cmake
      CMake Error: Not a file: /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/common/cmake/build-info-gen-cpp.cmake
      CMake Error: Error processing file: /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/common/cmake/build-info-gen-cpp.cmake
      [2/34] cd /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/vendor/llama.cpp/ggml/src && echo Embedding\ Metal\ library && sed -e '/#include "ggml-common.h"/r /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/ggml/src/ggml-common.h' -e '/#include "ggml-common.h"/d' < /private/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/pip-install-l07p5mxc/llama-cpp-python_f9d6911d530f4f4988c31d6dcbf10c06/vendor/llama.cpp/ggml/src/ggml-metal.metal > /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.metal && echo .section\ __DATA,__ggml_metallib > /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.s && echo .globl\ _ggml_metallib_start >> /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.s && echo _ggml_metallib_start: >> /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.s && echo .incbin\ \"/var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.metal\" >> /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.s && echo .globl\ _ggml_metallib_end >> /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.s && echo _ggml_metallib_end: >> /var/folders/6w/pcwbjyrx3w98xyksqn_y59pr0000gn/T/tmpjbiz25qi/build/autogenerated/ggml-metal-embed.s
      Embedding Metal library
      ninja: build stopped: subcommand failed.
      
      
      *** CMake build failed
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for llama-cpp-python
Failed to build llama-cpp-python
ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects

* Physical (or virtual) hardware you are using, e.g. for Linux:

`$ lscpu`

* Operating System, e.g. for Linux:

`$ uname -a`

* SDK version, e.g. for Linux:

```
$ python3 --version
$ make --version
$ g++ --version
```

# Failure Information (for bugs)

Please help provide information about the failure if this is a bug. If it is not a bug, please remove the rest of this template.

# Steps to Reproduce

Please provide detailed steps for reproducing the issue. We are not sitting in front of your screen, so the more detail the better.

1. step 1
2. step 2
3. step 3
4. etc.

**Note: Many issues seem to be regarding functional or performance issues / differences with `llama.cpp`. In these cases we need to confirm that you're comparing against the version of `llama.cpp` that was built with your python package, and which parameters you're passing to the context.**

Try the following:

1. `git clone https://github.com/abetlen/llama-cpp-python`
2. `cd llama-cpp-python`
3. `rm -rf _skbuild/` # delete any old builds
4. `python -m pip install .`
5. `cd ./vendor/llama.cpp`
6. Follow [llama.cpp's instructions](https://github.com/ggerganov/llama.cpp#build) to `cmake` llama.cpp
7. Run llama.cpp's `./main` with the same arguments you previously passed to llama-cpp-python and see if you can reproduce the issue. If you can, [log an issue with llama.cpp](https://github.com/ggerganov/llama.cpp/issues)

# Failure Logs

Please include any relevant log snippets or files. If it works under one configuration but not under another, please provide logs for both configurations and their corresponding outputs so it is easy to see where behavior changes.

Also, please try to **avoid using screenshots** if at all possible. Instead, copy/paste the console output and use [Github's markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) to cleanly format your logs for easy readability.

Example environment info:
```
llama-cpp-python$ git log | head -1
commit 47b0aa6e957b93dbe2c29d53af16fbae2dd628f2

llama-cpp-python$ python3 --version
Python 3.10.10

llama-cpp-python$ pip list | egrep "uvicorn|fastapi|sse-starlette|numpy"
fastapi                  0.95.0
numpy                    1.24.3
sse-starlette            1.3.3
uvicorn                  0.21.1

llama-cpp-python/vendor/llama.cpp$ git log | head -3
commit 66874d4fbcc7866377246efbcee938e8cc9c7d76
Author: Kerfuffle <44031344+KerfuffleV2@users.noreply.github.com>
Date:   Thu May 25 20:18:01 2023 -0600
```

**Comment by abetlen** [comment]
@mobeetle duplicate of #1662 which should be fixed now in 0.2.87

**Comment by HansvanHespen** [comment]
I got my problem solved by adding this line in my Dockerfile:"ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/local/cuda-12.5/compat""
